### PR TITLE
User create

### LINKED ISSUE #24: User-create
User::Class
create form
Validation account

Estimation : 7h

### PR DESCRIPTION
**Pull request by LFZDavid**

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+     *  )
+     */
+    private $name;
+    
+    /**
+     * @var string
+     * @Assert\NotNull
+     * @EqualTo(propertyPath="password", message="Vous n'avez pas tapé le même mot de passe")
+     */
+    private $confirm_password;
```
Comment (crazymad) [inline src/Entity/User.php:64]: camelCase

File: src/EventListener/CreateUserListener.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+use App\Entity\User;
+use App\Service\AccountValidator;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Csrf\TokenGenerator\TokenGeneratorInterface;
+
+class CreateUserListener
+{
+    private $passwordEncoder;
```
Comment (crazymad) [inline src/EventListener/CreateUserListener.php:13]: you can add strong typing here also

File: src/Entity/User.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+    /**
+     * @var bool
+     * @ORM\Column(type="boolean")
+     */
+    private $active;
```
Comment (crazymad) [inline src/Entity/User.php:70]: $active = false;

by default ? and then you can remove from the listener

File: src/Service/AccountValidator.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+namespace App\Service;
+
+use App\Entity\User;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class AccountValidator
+{
+    private $urlGenerator;
```
Comment (crazymad) [inline src/Service/AccountValidator.php:12]: strong typing

File: src/Service/AccountValidator.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        
+        try {
+            $this->mailer->send($email);
+        } catch (\Throwable $th) {
+            throw $th->getMessage();
+        }
+
+    }
+
+    protected function generateMailHtmlContent($url): string
```
Comment (crazymad) [inline src/Service/AccountValidator.php:42]: parameter string

**Review by crazymad — APPROVED** [review]
👍🏼

File: src/Entity/User.php (hunk 1 above)
Comment (LFZDavid) [reply src/Entity/User.php:64]: Fixed in user-login branch

File: src/EventListener/CreateUserListener.php (hunk 2 above)
Comment (LFZDavid) [reply src/EventListener/CreateUserListener.php:13]: Fixed in user-login branch

File: src/Entity/User.php (hunk 3 above)
Comment (LFZDavid) [reply src/Entity/User.php:70]: Fixed in user-login branch

File: src/Service/AccountValidator.php (hunk 4 above)
Comment (LFZDavid) [reply src/Service/AccountValidator.php:12]: Fixed in user-login branch

File: src/Service/AccountValidator.php (hunk 5 above)
Comment (LFZDavid) [reply src/Service/AccountValidator.php:42]: Fixed in user-login branch

**Pull request merged**